db.go

```diff
@@ -79,6 +81,16 @@ func Open(dirname string) (*DB, error) {
 	return db, nil
 }
 
+func (db *DB) Close() {
+	db.flushingChan <- db.memtables.mutable
+	db.memtables.mutable = NewMemtable(memtableSizeLimitBytes)
+	db.cancel()
+	db.wg.Wait()
+
+	close(db.flushingChan)
```

Comment:
### Runtime Errors
Handle potential runtime errors when closing channels. Specifically, ensure that `close(db.flushingChan)` and `close(db.compactionChan)` are checked for errors. Although `close` does not return an error, ensure that the channels are not nil before closing:
```go
if db.flushingChan != nil {
    close(db.flushingChan)
}
if db.compactionChan != nil {
    close(db.compactionChan)
}
```